transcom/mymove
# INT- B-19532-approved-advance-v3

## [B-19532](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A940488)

## Summary

**INTEGRATION**

This PR is a v3 followup to the original [PR](https://github.com/transcom/mymove/pull/12638).  An issue was discovered during PO demo where a PPM is in REJECT status and advance number is modified and SAVED, the status is automatically displayed as ACCEPT when PPM view is reloaded after SAVED. This means status of REJECTED was overwritten to EDITED/ACCEPT. After consulting with Dan Matthews, it was decided the UI shall programmatically select ACCEPT if user enters a new advance number regardless if REJECT is selected. It is assumed user's intent is to ACCEPT if he/she modifies the advance request number.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Steps to test: 

1. Log in as Services Counselor in Office and edit shipment on PPM

2. Enter an amount for Amount Requested and select Approve and hit Save & Continue

3. Edit Shipment again and edit the Amount Requested to different number but take note of saved value. Click REJECT radio button. Note the Advance Requested input reverts back to the persisted value.

4. Now, change advance number again, the status should automatically change to ACCEPT from REJECT. Modifying request in REJECT assumes user wants to ACCEPT before saving.

_Given I am a SC on the "Edit shipment details" page,
When I am reviewing the "Incentive & advance" area of the page,
Then when I update the advance amount,  Accept radio button is automatically selected if  Reject was previously selected._


![936799](https://github.com/transcom/mymove/assets/148127004/63a5595d-c31c-4c40-899c-d02545497085)

### Frontend
